lib/logstash/filters/fieldmap.rb

```diff
@@ -56,18 +60,61 @@ def filter(event)
       #  the key lenght, if not, explode
       split_src = event[@src_field].split(/#{@regex}/)
       @logger.debug? and @logger.debug("split_src is: ", :split_src => split_src)
+
+      if @text_qualifier
+        split_src_new_idx = split_src_orig_idx = 0
+        qualifier_start_idx = -1
+        split_src.map do |val|
+
+          val = val.strip   # this will cause problems if the text_qualifier is a space character
+
+          # Handle the case where we're in a qualified string
+          if val.start_with? @text_qualifier
+            qualifier_start_idx = split_src_orig_idx
+          elsif val.end_with? @text_qualifier
+            # TODO make this put in whatever it found to split on... not just a space
+            val = split_src[qualifier_start_idx..split_src_orig_idx].join(' ')
+            val = val.tr(@text_qualifier, "") # strips off the qualifier marks
+            qualifier_start_idx = -1
+          end
+
+          # this is the "not in the middle of a qualified string" (a.k.a normal) case
+          if qualifier_start_idx < 0
+            split_src[split_src_new_idx] = val
+            split_src_new_idx += 1
+          end
+
+          split_src_orig_idx += 1
+
+          # Handle the case where there is no "end" to the qualified string
+          if split_src_orig_idx >= split_src.length
+            event["tags"] ||= []
+            event["tags"] << @map_failure unless event["tags"].include?(@map_failure)
```

Comment:
can you change the error message here to be specific to the qualified string handling?